api/src/main/java/uk/gov/dhsc/htbhf/claimant/entity/constraint/EligibilityOverrideValidator.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.dhsc.htbhf.claimant.entity.constraint;
+
+import org.apache.commons.lang3.ObjectUtils;
+import uk.gov.dhsc.htbhf.claimant.entity.EligibilityOverride;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EligibilityOverrideValidator implements ConstraintValidator<ValidEligibilityOverride, EligibilityOverride> {
+    @Override
+    public void initialize(ValidEligibilityOverride constraintAnnotation) {
+        //No initialization required
+    }
+
+    @Override
+    public boolean isValid(EligibilityOverride eligibilityOverride, ConstraintValidatorContext context) {
+        if (null == eligibilityOverride) {
+            return true;
+        }
+
+        return ObjectUtils.allNotNull(
```

Comment:
Have you used this utility because you expect there to me more fields added to this? At the moment two simple `!= null` checks would be more concise and still quite readable.

Reply:
if the eligibility override is not null then none of the fields can be null. We will be adding third field List of childrenDob. We will add javadoc to understand it in better way

Reply:
I think adding JavaDoc with a couple of examples (as Apache Commons does very well) would really help here.